ArduPlane/GCS_Mavlink.cpp

```diff
@@ -372,6 +372,26 @@ void Plane::send_nav_controller_output(mavlink_channel_t chan)
         nav_controller->crosstrack_error());
 }
 
+void Plane::send_position_target_global_int(mavlink_channel_t chan)
+{
+    mavlink_msg_position_target_global_int_send(
+        chan,
+        AP_HAL::millis(), // time_boot_ms
+        MAV_FRAME_GLOBAL_INT, // targets are always global altitude
+        0xFFF8, // ignore everything except the x/y/z components
+        next_WP_loc.lat, // latitude as 1e7
+        next_WP_loc.lng, // longitude as 1e7
+        next_WP_loc.alt / 100.0f, // altitude is sent as a float
```

Comment:
No, but I'll change it for you :)

On Wed, Apr 13, 2016 at 11:10 PM, Tom Pittenger <notifications@github.com>
wrote:

> In ArduPlane/GCS_Mavlink.cpp
> <https://github.com/ArduPilot/ardupilot/pull/3899#discussion_r59666995>:
>
> > @@ -372,6 +372,26 @@ void Plane::send_nav_controller_output(mavlink_channel_t chan)
> >          nav_controller->crosstrack_error());
> >  }
> >
> > +void Plane::send_position_target_global_int(mavlink_channel_t chan)
> > +{
> > +    mavlink_msg_position_target_global_int_send(
> > +        chan,
> > +        AP_HAL::millis(), // time_boot_ms
> > +        MAV_FRAME_GLOBAL_INT, // targets are always global altitude
> > +        0xFFF8, // ignore everything except the x/y/z components
> > +        next_WP_loc.lat, // latitude as 1e7
> > +        next_WP_loc.lng, // longitude as 1e7
> > +        next_WP_loc.alt / 100.0f, // altitude is sent as a float
>
> Has anyone done any metrics comparing float divide versus multiply on
> pixhawk? I always cringe when I see /100 instead of *0.01
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/ArduPilot/ardupilot/pull/3899/files/d6ef06b2dc5195fd04f760ffaf621fa69f2ca560#r59666995>
>